did_core/did_parser_nom/tests/did/positive.rs

```diff
@@ -0,0 +1,93 @@
+use did_parser_nom::Did;
+
+macro_rules! test_cases_positive {
+    ($($name:ident: $input:expr, $expected_did:expr, $expected_method:expr, $expected_namespace:expr, $expected_id:expr)*) => {
+        $(
+            #[test]
+            fn $name() {
+                println!("Testing {}", $input);
+                let parsed_did = Did::parse($input.to_string()).unwrap();
+
+                assert_eq!(parsed_did.did(), $expected_did, "DID");
+                assert_eq!(parsed_did.method(), $expected_method, "Method");
+                assert_eq!(parsed_did.namespace(), $expected_namespace, "Namespace");
+                assert_eq!(parsed_did.id(), $expected_id, "ID");
+            }
+        )*
+    };
+}
+
+test_cases_positive! {
+    test_case1:
+        "did:example:123456789abcdefghi",
+        "did:example:123456789abcdefghi",
```

Comment:
given every test case have the first 2 values identical, it seems like redundancy

Reply:
I agree with the statement, it seems redundant, but the input and expected output of a method are conceptually distinct (e.g. can change independently).